src/components/patient/Orders.js

```diff
@@ -1,34 +1,429 @@
+  // can even further be optimized to check if openMed == true, ONLY then update
+
+  // previously used to display json med list: <pre>{JSON.stringify(data, null, 4)}</pre>
 
   return (
     <Box>
-      <TextField
-        label="Search"
-        variant="outlined"
-        value={value}
-        onChange={(x) => setValue(x.target.value)}
-      />
-      <Button variant="outlined" onClick={() => { setOpen(!open) }}>
-        View
-      </Button>
-      <Dialog onClose={() => { setOpen(false) }} open={open}>
-        <pre>{JSON.stringify(data, null, 4)}</pre>
-      </Dialog>
+      <div>
+        <TextField
+          label="Search Medications"
+          variant="outlined"
+          value={value}
+          onChange={(x) => setValue(x.target.value)}
+        />
+        <Button variant="outlined" onClick={() => { setOpenMed(!openMed) }}>
+          Add Medication
+        </Button>
+        <Button variant="outlined" onClick={() => { setOpenLab(!openLab) }}>
+          Add Lab
+        </Button>
+
+        <Dialog fullWidth maxWidth="md" onClose={() => { setOpenMed(false) }} open={openMed}>
+
+          <nav>
+            <List>
+              {data.map((m) => (
+                <ListItem disablePadding>
+                <ListItemButton onClick={() => 
+                  {tempMedChange(m); 
+                  setRoute(m.fields.route ? m.fields.route[0] : route); 
+                  setDose(m.fields.dose ? m.fields.dose[0][0] : dose); 
+                  setFreq(m.fields.frequency ? m.fields.frequency : freq); 
+                  setRefill(m.fields.refills ? m.fields.refills : refill); 
+                  setOpenDose(!openDose); 
+                  setOpenMed(false);}}>
+                  <ListItemText primary={m.name}/>
+                </ListItemButton>
+              </ListItem>
+              ))}
+            </List>
+          </nav>
+        </Dialog>
+
+
+        <Dialog fullWidth maxWidth="md" onClose={() => { setOpenDose(false) }} open={openDose}>
+                
+        <div style={{backgroundColor: "blue", height: "40px"}}>
+            {tempMed ? tempMed.name : ''}
+            <DialogActions style={{float: "right"}}>
+              <Button onClick={() => { closeDose(true) }}><CheckIcon color="success"/>Accept</Button>
+              <Button onClick={() => { closeDose(false) }}><ClearIcon color="error"/>Cancel</Button>
+            </DialogActions>
+        </div>
+                <p>Route: </p> 
+                <ToggleButtonGroup
+                  value={route}
+                  exclusive
+                  onChange={routeChange}
+                >
+                {
+                tempMed ? tempMed.fields.route.map((m) => (
+                  createToggleButton(m, m)
+                )) : ''
+                }
+              </ToggleButtonGroup>
+
+              <p>Dose: </p> 
+                <ToggleButtonGroup
+                  value={dose}
+                  exclusive
+                  onChange={doseChange}
```

Comment:
You can just inline these instead of making new functions for each one! Example: 
```jsx

ToggleButtonGroup
  value={dose}
  exclusive
  onChange={(event, value) => setDose(value)}
>
```